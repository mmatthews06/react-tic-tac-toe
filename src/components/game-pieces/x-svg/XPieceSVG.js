
import React from 'react';
import PropTypes from 'prop-types';

import EmptyPieceSVG from '../empty-svg/EmptyPieceSVG';

import './XPieceSVG.css';

export default function XPieceSVG({ margin }) {
  const endMargin = 100 - margin;

  return (
    <svg x="0" y="0" height="100%" width="100%" viewBox="0 0 100 100" className="x-piece-svg">
      <EmptyPieceSVG />
      <line x1={`${margin}%`} y1={`${margin}%`} x2={`${endMargin}%`} y2={`${endMargin}%`} />
      <line x1={`${endMargin}%`} y1={`${margin}%`} x2={`${margin}%`} y2={`${endMargin}%`} />
    </svg>
  );
}

XPieceSVG.propTypes = {
  margin: PropTypes.number,
};

XPieceSVG.defaultProps = {
  margin: 25,
};
