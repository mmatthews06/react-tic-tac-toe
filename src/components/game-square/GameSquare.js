
import React from 'react';
import PropTypes from 'prop-types';

import EmptyPieceSVG from '../game-pieces/empty-svg/EmptyPieceSVG';

export default function GameSquare({ x, y, height, width, position, children, playerMoveHandler }) {
  return (
    <svg
      id={`${position}-square`}
      x={x}
      y={y}
      height={height}
      width={width}
      onClick={() => playerMoveHandler(position)}
    >
      {children}
    </svg>
  );
}

GameSquare.propTypes = {
  children: PropTypes.element,
  position: PropTypes.number.isRequired,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  playerMoveHandler: PropTypes.func.isRequired,
};

GameSquare.defaultProps = {
  children: <EmptyPieceSVG />,
};
