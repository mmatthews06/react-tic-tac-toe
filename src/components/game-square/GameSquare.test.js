/* global describe, expect, it */
import React from 'react';
import { shallow } from 'enzyme';

import GameSquare from './GameSquare';
import EmptyPieceSVG from '../game-pieces/empty-svg/EmptyPieceSVG';
import XPieceSVG from '../game-pieces/x-svg/XPieceSVG';
import OPieceSVG from '../game-pieces/o-svg/OPieceSVG';

const props = {
  x: 0,
  y: 0,
  position: 0,
  height: 167,
  width: 167,
  playerMoveHandler: () => {},
};

describe('GameSquare', () => {
  describe('Child Piece Rendering', () => {
    it('should render an EmptyPieceSVG by default', () => {
      expect(shallow(<GameSquare {...props} />).find(EmptyPieceSVG)).toHaveLength(1);
    });

    it('should render an XPieceSVG instead of default if passed as a child', () => {
      const wrapper = shallow(
        <GameSquare {...props}>
          <XPieceSVG />
        </GameSquare>,
      );
      expect(wrapper.find(EmptyPieceSVG)).toHaveLength(0);
      expect(wrapper.find(XPieceSVG)).toHaveLength(1);
    });

    it('should render an OPieceSVG instead of default if passed as a child', () => {
      const wrapper = shallow(
        <GameSquare {...props}>
          <OPieceSVG />
        </GameSquare>,
      );
      expect(wrapper.find(EmptyPieceSVG)).toHaveLength(0);
      expect(wrapper.find(OPieceSVG)).toHaveLength(1);
    });
  });

  it('should match a default snapshot', () => {
    expect(shallow(<GameSquare {...props} />)).toMatchSnapshot();
  });

  it('should match a snapshot with a child', () => {
    expect(shallow(
      <GameSquare {...props}>
        <OPieceSVG />
      </GameSquare>,
    )).toMatchSnapshot();
  });
});
