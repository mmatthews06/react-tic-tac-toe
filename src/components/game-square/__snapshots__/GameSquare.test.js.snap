// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameSquare should match a default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameSquare
    height={167}
    position="top-left"
    width={167}
    x={0}
    y={0}
>
    <EmptyPieceSVG />
</GameSquare>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <EmptyPieceSVG />,
      "height": 167,
      "id": "top-left-square",
      "onClick": [Function],
      "width": 167,
      "x": 0,
      "y": 0,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <EmptyPieceSVG />,
        "height": 167,
        "id": "top-left-square",
        "onClick": [Function],
        "width": 167,
        "x": 0,
        "y": 0,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GameSquare should match a snapshot with a child 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameSquare
    height={167}
    position="top-left"
    width={167}
    x={0}
    y={0}
>
    <OPieceSVG
        margin={25}
    />
</GameSquare>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <OPieceSVG
        margin={25}
/>,
      "height": 167,
      "id": "top-left-square",
      "onClick": [Function],
      "width": 167,
      "x": 0,
      "y": 0,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "margin": 25,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <OPieceSVG
          margin={25}
/>,
        "height": 167,
        "id": "top-left-square",
        "onClick": [Function],
        "width": 167,
        "x": 0,
        "y": 0,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "margin": 25,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
